# Prune for LD with SNPfiltR

library(vcfR)
library(tidyverse)
library(adegenet)
library(SNPfiltR)
library(StAMPP)
library(gplots)

# these can been 
vcfR<-read.vcfR("data/stacks_filtered_pre_LD_prune/populations.snps.vcf")

# read in sample info
samples<-read_csv("data/master_genetic_samples.csv")

# make sure sample files matches samples in vcf
samples$Unique_sample_ID==colnames(vcfR@gt)[-1]
# they don't but that's because of the '_2nd_seq' samples


hard_filter(vcfR=vcfR, depth = 3, gq = 30)
vcf_thin<-distance_thin(vcfR, min.distance = 500)
vcfR::write.vcf(vcf_thin, file="output/SNPfiltR/distance_thin.snps.vcf.gz")


#convert to genlight
gen<-vcfR2genlight(vcf_thin)
#assign populations (a StaMPP requirement)
gen@pop<-as.factor(gen@ind.names)
#generate pairwise divergence matrix
sample.div <- stamppNeisD(gen, pop = FALSE)

# output in phylip format
stamppPhylip(sample.div, file="output/SNPfiltR/thinned.phy.dst")

heatmap.2(sample.div)
plot(nj(sample.div))


# try filtering with more than just distance thin to see if there is a big difference
# this is totally unfiltered
dat<-read.vcfR("data/unfiltered_vcf/populations.snps.vcf.gz")

# hard filter to minimum depth of 3, and minimum genotype quality of 30
dat<-hard_filter(vcfR=dat, depth=3, gq=30)

# filter for allelic balance
dat<-filter_allele_balance(dat, min.ratio = 0.10, max.ratio = 0.90)

# Checked coverage depth with max_depth()
dat<-max_depth(dat, maxdepth = 100)

# minor allele count
dat<-min_mac(dat, min.mac = 1)

# filter by proportion missing data by sample
# ran missing_by_sample(dat)
dat<-missing_by_sample(dat, cutoff=0.7)

#remove invariant sites generated by sample trimming
dat<-min_mac(dat, min.mac = 1)

# stop and check what the neighbor-joining tree looks like?
# #convert to genlight
# dat.gen<-vcfR2genlight(dat)
# #assign populations (a StaMPP requirement)
# dat.gen@pop<-as.factor(dat.gen@ind.names)
# #generate pairwise divergence matrix
# dat.sample.div <- stamppNeisD(dat.gen, pop = FALSE)
# 
# # output in phylip format
# stamppPhylip(dat.sample.div, file="output/SNPfiltR/no_stacks_filters/snpfiltr_partial.phy.dst")


# remove overlapping snps
#generate dataframe containing information for chromosome and bp locality of each SNP
df<-as.data.frame(dat@fix[,1:2])
#calc number of duplicated SNPs to remove
nrow(df) - length(unique(paste(df$CHROM,df$POS)))
# 1532 SNPs

# removed duplicated SNPs
dat<-dat[!duplicated(paste(df$CHROM, df$POS)),]

# I guess I'll filter on SNP completeness to go along with the rest of the workflow
dat_thinned<-missing_by_snp(dat, cutoff=0.9)


# check plots of depth and genotype quality (this took a long time)
#plot depth per snp and per sample
#depth
# dp <- extract.gt(dat_thinned, element = "DP", as.numeric=TRUE)
# heatmap.bp(dp, rlabels = FALSE)
# 
# #plot genotype quality per snp and per sample
# gq <- extract.gt(dat_thinned, element = "GQ", as.numeric=TRUE)
# heatmap.bp(gq, rlabels = FALSE)

# I think I'll want to write out 3 files:
# 1) the vcf before the 90% completeness and without LD pruning-> can use 
# this for Fst with stammp to complete to with 90% SNP completeness
vcfR::write.vcf(dat_thinned, 
        file="output/SNPfiltR/vcfs/stacks_unfiltered_90per_snp_complete_no_distance_LD_thin/")


# 2) the vcf after 90% completeness
# 3) the vcf after 90% completeness and after LD distance thinning

# filter for linkage


                                     
